lib/npa.js

```diff
@@ -87,6 +88,22 @@ function resolve (name, spec, where, arg) {
   }
 }
 
+function toPurl (arg, reg) {
+  const res = npa(arg)
+
+  if (res.type !== 'version') {
+    throw invalidPurlType(res.type, res.raw)
+  }
+
+  // URI-encode leading @ of scoped packages
+  let purl = 'pkg:npm/' + res.name.replace(/^@/, '%40') + '@' + res.rawSpec
+  if (reg) {
```

Comment:
```suggestion
  if (reg !== defaultRegistry) {
```
this ensures that if someone explicitly passes the default, it'll be omitted